src/esp/sim/Simulator.h

```diff
@@ -924,6 +942,9 @@ class Simulator {
   SimulatorConfiguration config_;
 
   std::vector<agent::Agent::ptr> agents_;
+  // SensorSuite of all sensors available to the simulator
+  esp::sensor::SensorSuite sensorSuite_;
```

Comment:
I do not think you need this `sensorSuite_`. Maybe you can have one defined in the viewer.cpp for your demo, but not in the simulator.
@aclegg3 : How do you think? I remembered that when we discuss this project, you mentioned once that we might need such a global sensorSuite to collect all the sensors? Can you think of any usage case where such a global `sensorSuite` is a must-have?

Reply:
Python uses its own for backwards compatability, but there is no reason if you can quickly and easily get all the sensors in the scene.

Reply:
Thanks for the comments - I included a global `sensorSuite_` into Simulator to keep track of all the sensors. I thought about just keeping a pointers to the last added sensor, but later reasoned that it would be useful to be able to get all the sensors, not just the agent sensor and the last added sensor. 

I'm happy to change this, but right now it seems like it's not so easy to get all the sensors in the scene. 

Let me know any thoughts or if there is a better way to keep track of all the sensors! :)

Reply:
The question is that why you need to get all the sensors in the scene? 
When you add a sensor to an object, the "add" function should give a pointer to this sensor, and you keep it somewhere so that you can use it later.

I am not against to have such a sensorSuite in the viewer.cpp for example, but not here, in the simulator class.